gremlin-go/driver/graphBinary.go

```diff
@@ -577,6 +581,53 @@ func pathReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return p, nil
 }
 
+func timeWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	t := value.(time.Time)
+	_, err := typeSerializer.write(t.UnixMilli(), buffer)
+	if err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+func timeReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) (interface{}, error) {
+	newTime, err := typeSerializer.read(buffer)
+	if err != nil {
+		return nil, err
+	}
+	return time.UnixMilli(newTime.(int64)), nil
+}
+
+func durationWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	t := value.(time.Duration)
+	sec := int64(t.Seconds())
+	millis := t.Seconds() - float64(sec)
```

Comment:
Why did you convert Seconds, which is a float64 to int64, and then convert it bakc into float64, in order to subtract it from itself?

Reply:
The format for Duration on Tinkerpop is `Format: {seconds}{nanos}`
I will add this as a comment before the writer function. The reason I did it like that was to convert pure nanoseconds to a combination of seconds + nanoseconds.